controllers/networktopology.go

```diff
@@ -0,0 +1,113 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/giantswarm/aws-network-topology-operator/pkg/registrar"
+	"github.com/giantswarm/microerror"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util/annotations"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const FinalizerNetTop = "network-topology.finalizers.giantswarm.io"
+
+//counterfeiter:generate . ClusterClient
+type ClusterClient interface {
+	Get(context.Context, types.NamespacedName) (*capi.Cluster, error)
+	AddFinalizer(context.Context, *capi.Cluster, string) error
+	RemoveFinalizer(context.Context, *capi.Cluster, string) error
+}
+
+//counterfeiter:generate . Registrar
+type Registrar interface {
+	Register(context.Context, *capi.Cluster) error
+	Unregister(context.Context, *capi.Cluster) error
+}
```

Comment:
Why use the `Registrar` approach? will this operator create different resources?

Reply:
Possibly. I've been focussing on the GiantSwarmManaged mode for this which is pretty straight forward. When it comes to the UserManaged mode it's possible there's more we need to handle but I'm not sure at this stage.

Are there any strong reasons not to use it at this stage? Or do you think I should refactor it for now and add in when needed at a later date?